gtsam/slam/dataset.h

```diff
@@ -221,7 +221,7 @@ struct SfmTrack {
   SfmTrack(): p(0,0,0) {}
   SfmTrack(const gtsam::Point3& pt) : p(pt) {}
```

Comment:
@varunagrawal @ayushbaid perhaps we should do
```
SfmTrack(r = 0, g = 0, b = 0): p(0,0,0), r(r), g(g), b(b) {}
SfmTrack(const gtsam::Point3& pt, r = 0, g = 0, b = 0) : p(pt), r(r), g(g), b(b) {}
```
so we can use the RGB values later in rendering?

Reply:
perhaps `unsigned char` is more appropriate?

Reply:
`size_t`

Reply:
Actually keeping it float makes sense in case the values aren't in the 0-255 range.

Reply:
I would say it's probably better to rethink this a bit: what if the image is 12bit per channel? What if the input is grayscale? What are the implications of assuming 8bit RGB?

Reply:
totally fair, good points

Reply:
@ProfFan maybe we can discuss this a bit more -- I do think that 8bit is much more common for our use cases in GTSAM and GTSFM, and I also think that if the input is grayscale, they wouldn't use the RGB attributes in the first place. And I think this attribute is purely for 3d point cloud visualization purposes, and I don't think grayscale values would contribute meaningfully to a 3d visualization.

Reply:
Agree with John. As this is just used for visualization, we can just stick with float R, G, B values, normalized to [0, 1] range.